gateway/gateway.go

```diff
@@ -454,9 +454,9 @@ func (gw *Gateway) SendMessage(
 		msg.Channel = rmsg.Channel
 	}
 
-	msg.ParentID = gw.getDestMsgID(rmsg.Protocol+" "+canonicalParentMsgID, dest, channel)
+	msg.ParentID = gw.getDestMsgID(canonicalParentMsgID, dest, channel)
```

Comment:
not sure exactly what's going on here

The previous code caused multiple protocol names to be included in the message ID, which seems wrong

Also removed the protocol from the message ID in the fallback case since that gets sent to the destination as invalid